### PR TITLE
Add --disable-sgmv flag

### PR DESCRIPTION
**Pull request by joseph-predibase**
Adds --disable-sgmv, which sets `DISABLE_SGMV=1`. 

Tested by spinning up a deployment with this build and using debug code to verify that the env var is seen by the lorax server.

### CODE REVIEW

File: server/lorax_server/utils/sgmv.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
     import punica_kernels as _kernels
 
     HAS_SGMV = not bool(os.environ.get("DISABLE_SGMV", ""))
+    print("HAS_SGMV: ", HAS_SGMV)
```
Comment (tgaddair) [inline server/lorax_server/utils/sgmv.py:13]: Please remove debug code.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 name: Build and push docker image to ghcr
 
 on:
+  pull_request:
```
Comment (tgaddair) [inline .github/workflows/build.yaml:4]: Please remove before landing.

**Review by tgaddair — APPROVED** [review]
LGTM!

File: server/lorax_server/utils/sgmv.py (hunk 1 above)
Comment (joseph-predibase) [reply server/lorax_server/utils/sgmv.py:13]: still a draft! was building to deploy and make sure it was propagated, i can't get it to build on my laptop so...

**Comment by joseph-predibase** [comment]
@tgaddair getting a bunch of GPG errors when trying to fetch some nvidia stuff, any idea what's up with that? Doesn't seem remotely related to my PR. (Build log here: https://github.com/predibase/lorax/actions/runs/11338623040/job/31532139387?pr=639)

This is a pretty trivial PR but would be nice to test it...

**Comment by tgaddair** [comment]
I think the issue is with the builder instance running out of space. I'll take a look and see if I can clean it up.

**Pull request merged**